blueflood-core/src/main/java/com/rackspacecloud/blueflood/tools/ops/FixTTL.java

```diff
@@ -0,0 +1,118 @@
+package com.rackspacecloud.blueflood.tools.ops;
+
+import org.apache.commons.cli.*;
+import com.netflix.astyanax.model.*;
+import com.rackspacecloud.blueflood.io.*;
+import com.rackspacecloud.blueflood.types.*;
+import java.util.*;
+
+public class FixTTL {
+    // Update all the TTL's for a set of metrics over a particular time range
+    public static void fixTTLs(ColumnFamily CF, String tenantID,
+                               List<String> metrics, Range range, Integer newTTL)
+            throws Exception {
+
+        List<Locator> locators = new ArrayList<Locator>();
+        for (String path: metrics) {
+            locators.add(Locator.createLocatorFromPathComponents(tenantID, path));
+        }
+
+        // Print out current TTL's, and update them
+        Map<Locator, ColumnList<Long>> data =
+                AstyanaxReader.getInstance().getColumnsFromDB(locators, CF, range);
+
+        for (Map.Entry<Locator, ColumnList<Long>> entry: data.entrySet()) {
+            Locator l = entry.getKey();
+            System.out.println("Current Locator is " + l.toString());
+            ColumnList<Long> cols = entry.getValue();
+
+            System.out.println("Number of cols is " + cols.size());
+            for (Column<Long> col: cols) {
+                System.out.println("col name is " + col.getName() + " ttl is " + col.getTtl());
+            }
+            AstyanaxWriter.getInstance().updateTTL(CF, l, cols, newTTL);
+        }
+
+        System.out.println("\n\nThe updated TTL's are:");
+        data = AstyanaxReader.getInstance().getColumnsFromDB(locators, CF, range);
+        for (Map.Entry<Locator, ColumnList<Long>> entry: data.entrySet()) {
+            Locator l = entry.getKey();
+            System.out.println("Locator of updated metric is " + l.toString());
+            ColumnList<Long> cols = entry.getValue();
+            for (Column<Long> col: cols) {
+                System.out.println("col name is " + col.getName() + " new ttl is " + col.getTtl());
+            }
+        }
+        System.out.println("FixTTL done");
+    }
+
+    public static void main(String args[]) throws Exception {
+        Map<String, Object> options = OptionsHandler.parseOptions(args);
+        ColumnFamily CF = CassandraModel.CF_METRICS_FULL;
```

Comment:
Eventually, it will be useful to have the tool support the functionality to fix ttls for rollup metrics as well. I think even right now people can supply there own ttls, which will be applied to the rollup granularities, but I see some some use case in having the ability to change the ttls for rollup metrics as well, like may be someone wants to preserve some critical metrics for a longer period of time, without losing too much data when it gets rolled up to higher granularity.